src/include/gurt/debug_setup.h

```diff
@@ -0,0 +1,206 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef __GURT_DEBUG_SETUP_H__
+#define __GURT_DEBUG_SETUP_H__
+
+#include <gurt/dlog.h>
+/**
+ * \file
+ * Debug macros and functions
+ */
+
+/** @addtogroup GURT_DEBUG
+ * @{
+ */
+#define DD_FAC(name)	(d_##name##_logfac)
+
+#define _D_LOG_FAC_DECL(name)	DD_FAC(name)
+
+#ifndef D_LOGFAC
+#define D_LOGFAC	DD_FAC(misc)
+#endif
+
+/** Arguments to priority bit macros are
+ *      flag            Variable name of the priority bit flag
+ *      s_name          Short name of the flag
+ *      l_name          Long name of the flag
+ *      default_mask    Should always be 0 for debug bits
+ *      arg             Argument passed along.  Use D_NOOP when not required
+ */
+#define D_FOREACH_GURT_DB(ACTION, arg)          \
+	/** All debug streams */                \
+	ACTION(DB_ALL,   all,   all,   0, arg)  \
+	/** Generic debug stream */             \
+	ACTION(DB_ANY,   any,   any,   0, arg)  \
+	/** Extremely verbose debug stream */   \
+	ACTION(DB_TRACE, trace, trace, 0, arg)  \
+	/** Memory operations */                \
+	ACTION(DB_MEM,   mem,   mem,   0, arg)  \
+	/** Network operations */               \
+	ACTION(DB_NET,   net,   net,   0, arg)  \
+	/** I/O operations */                   \
+	ACTION(DB_IO,    io,    io,    0, arg)  \
+	/** Test debug stream */                \
+	ACTION(DB_TEST,  test,  test,  0, arg)
+
+/* These macros are intended to be used with FOREACH macros that define
+ * debug bits for your library.   See D_FOREACH_GURT_DB for an example.  Not
+ * all fields are used in individual macros.
+ *
+ * These macros can also be used standalone but are not quite as convenient
+ * due to required but unused arguments
+ */
+
+/** Declare an extern for a debug flag variable
+ *
+ * \param flag		The name of the variable used to store the value of the
+ *			debug bit.
+ * \param s_name	Unused
+ * \param l_name	Unused
+ * \param mask		Unused
+ * \param arg		Unused
+ */
+#define D_LOG_DECLARE_DB(flag, s_name, l_name, mask, arg)	\
+	extern d_dbug_t	flag;
+
+/** Instantiate a debug flag variable
+ *
+ * \param flag		The name of the variable used to store the value of the
+ *			debug bit.
+ * \param s_name	Unused
+ * \param l_name	Unused
+ * \param mask		Unused
+ * \param arg		Unused
+ */
+#define D_LOG_INSTANTIATE_DB(flag, s_name, l_name, mask, arg)	\
+	d_dbug_t	flag;
+
+/** Internal use macro as part of D_LOG_REGISTER_DB.  It's multiline on purpose
+ *  so we can break from the foreach loop.  Checkpatch will complain but don't
+ *  see a better way.
+ */
+#define _D_LOG_ALLOCATE_DBG_BIT(dbgbit, s_name, l_name, mask, rc)	\
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop